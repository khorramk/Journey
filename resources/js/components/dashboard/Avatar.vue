<template>
    <div>
        <div class="base-image-input" :style="{ 'background-image': `url(${imageData}`}" @click="chooseImage">
            <span>
                choose image
            </span>
            <input type="file" name="" ref="fileinput" id="" @input="onSelectFile">
        </div>
<!--
            <input id="imageUpload"   v-on:click="clicked($event)" type="file"
            name="profile_photo" placeholder="Photo" required="" capture>
            <input type="image" :src="pathToImage" alt=""> -->



    </div>
</template>

<script>
    export default {
        data() {
            return {
                image: '',
               imageData: null

            }
        },
        methods: {
            clicked(event){
                console.log(event.target.value);
                this.$data.pathToImage = event.target.value;
            //     const fileReader = new FileReader();
            //    fileReader.onload = (event)=>{
            //        console.log(event.target.files['0'].file)
            //    }

            //    fileReader.readAsBinaryString(event.target.files['0'].file);
                // console.log(this.$refs.file.attributes.src);
                // console.log(event.target.files);
                // if(event.target.files['0'].name === undefined){

                // }else{
                //     this.$refs.file.attributes.src= event.target.files['0'].name;
                // }
                // this.$refs.file.attributes.src.nodeValue = event.files['0'].file.name
            },
            insertAvatar() {
                submit
                axios.post('/api/avatar').then((resp)=> console.log('avatar uploaded'))
            },
            onSelectFile(event){
                this.$data.imageData = event.target.value;
            },
            chooseImage(){
                this.$refs.fileInput;
            }
        },
    }

</script>

<style lang="scss" scoped>

</style>
