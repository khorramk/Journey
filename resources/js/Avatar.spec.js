import { mount } from '@vue/test-utils';
import Vue from 'vue';
import axios from 'axios';
import VueRouter from 'vue-router';
import Avatar from './components/dashboard/Avatar';
import expectExport from 'expect';


const vm = new Vue(Avatar);

it('renders the elements', () => {

    vm.$nextTick().then(()=> {

     expect(vm).toMatchSnapshot();
    }).catch().finally()
  })
