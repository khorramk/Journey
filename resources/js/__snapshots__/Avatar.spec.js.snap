// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the elements 1`] = `
Vue {
  "$attrs": Object {},
  "$children": Array [],
  "$createElement": [Function],
  "$listeners": Object {},
  "$options": Object {
    "_base": [Function],
    "components": Object {},
    "data": [Function],
    "directives": Object {},
    "filters": Object {},
    "methods": Object {
      "insertAvatar": [Function],
      "onSelectFile": [Function],
      "openSelection": [Function],
      "removeSelection": [Function],
    },
    "render": [Function],
    "staticRenderFns": Array [],
  },
  "$parent": undefined,
  "$refs": Object {},
  "$root": [Circular],
  "$scopedSlots": Object {},
  "$slots": Object {},
  "$vnode": undefined,
  "_c": [Function],
  "_data": Object {
    "closeUponSelectImage": true,
    "file": "",
    "imageData": "",
    "switch": false,
  },
  "_directInactive": false,
  "_events": Object {},
  "_hasHookEvent": false,
  "_inactive": null,
  "_isBeingDestroyed": false,
  "_isDestroyed": false,
  "_isMounted": false,
  "_isVue": true,
  "_renderProxy": Vue {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_base": [Function],
      "components": Object {},
      "data": [Function],
      "directives": Object {},
      "filters": Object {},
      "methods": Object {
        "insertAvatar": [Function],
        "onSelectFile": [Function],
        "openSelection": [Function],
        "removeSelection": [Function],
      },
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {
      "closeUponSelectImage": true,
      "file": "",
      "imageData": "",
      "switch": false,
    },
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": false,
    "_isVue": true,
    "_renderProxy": [Circular],
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": null,
    "_watcher": null,
    "_watchers": Array [],
    "closeUponSelectImage": true,
    "file": "",
    "imageData": "",
    "insertAvatar": [Function],
    "onSelectFile": [Function],
    "openSelection": [Function],
    "removeSelection": [Function],
    "switch": false,
  },
  "_self": [Circular],
  "_staticTrees": null,
  "_uid": 0,
  "_vnode": null,
  "_watcher": null,
  "_watchers": Array [],
  "closeUponSelectImage": true,
  "file": "",
  "imageData": "",
  "insertAvatar": [Function],
  "onSelectFile": [Function],
  "openSelection": [Function],
  "removeSelection": [Function],
  "switch": false,
}
`;
